File: ENTIDADES/comercial/Oferta.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ENTIDADES.comercial
{
    [Table("Oferta", Schema = "Comercial")]

    public class Oferta:AuditoriaColumna
    {
        [Key]
        public int idoferta { get; set; }
        public string estado { get; set; }
        public string nombre { get; set; }
        public DateTime?    fechainicio { get; set; }
        public DateTime?    fechafin { get; set; }
        public string tipo { get; set; }
        public bool? reqcliente { get; set; }
        public bool? grabarcosto0 { get; set; }
        [NotMapped]
        public string jsondetalle { get; set; }
        [NotMapped]
        public string jsonobsequios { get; set; }
    }
}

File: ERP/wwwroot/js/modelos/comercial/Oferta.js
class Oferta {
    constructor() {
        this.idoferta = 0;
        this.estado = 'HABILITADO';
        this.nombre = '';
        this.fechainicio = '';
        this.fechafin = '';
        this.tipo = '';
        this.reqcliente = false;
        this.grabarcosto0 = false;
    }
   
}
class DetalleOferta {
    constructor() {
        this.iddetalle = 0;
        this.idoferta = 0;
        this.idproducto = 0;
        this.isfraccion = false;
        this.cantidad = 0;
        this.monto = 0;
        this.estado = "HABILITADO";
    }   
}
class ObsequioOferta {
    constructor() {
        this.idobsequio = 0;
        this.idoferta = 0;
        this.idproducto = 0;
        this.cantidad = 0;
        this.descuento = 0;
        this.isfraccion = false;
        this.estado = "HABILITADO";

    }
   
}